Reading Data

In [4]:
import pandas as pd
import numpy as np

In [6]:
ds = pd.read_csv('customer churn data.csv')
ds.head(3)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0   CUST0000    Male              0      No        Yes      23           No   
1   CUST0001  Female              0     Yes         No      43           No   
2   CUST0002    Male              1      No         No      51          Yes   

      MultipleLines InternetService       OnlineSecurity  ...  \
0  No phone service              No  No internet service  ...   
1  No phone service             DSL                  Yes  ...   
2                No             DSL                   No  ...   

      DeviceProtection          TechSupport          StreamingTV  \
0  No internet service  No internet service  No internet service   
1                  Yes                   No                  Yes   
2                  Yes                  Yes                   No   

       StreamingMovies        Contract PaperlessBilling     PaymentMethod  \
0  No internet service  Month-to-month              Yes     Bank transfer   
1                   No  Month-to-month               No      Mailed check   
2                   No        One year               No  Electronic check   

  MonthlyCharges  TotalCharges  Churn  
0          49.85       1146.55     No  
1         100.70       4330.10    Yes  
2          97.33       4963.83    Yes  

[3 rows x 21 columns]

In [7]:
ds.shape

(5880, 21)

In [8]:
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5880 entries, 0 to 5879
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        5880 non-null   object 
 1   gender            5880 non-null   object 
 2   SeniorCitizen     5880 non-null   int64  
 3   Partner           5880 non-null   object 
 4   Dependents        5880 non-null   object 
 5   tenure            5880 non-null   int64  
 6   PhoneService      5880 non-null   object 
 7   MultipleLines     5880 non-null   object 
 8   InternetService   5880 non-null   object 
 9   OnlineSecurity    5880 non-null   object 
 10  OnlineBackup      5880 non-null   object 
 11  DeviceProtection  5880 non-null   object 
 12  TechSupport       5880 non-null   object 
 13  StreamingTV       5880 non-null   object 
 14  StreamingMovies   5880 non-null   object 
 15  Contract          5880 non-null   object 
 16  PaperlessBilling  5880 non-null   object 


Missing Values

In [11]:
(ds.isnull().sum()/ds.shape[0])*100

customerID          0.0
gender              0.0
SeniorCitizen       0.0
Partner             0.0
Dependents          0.0
tenure              0.0
PhoneService        0.0
MultipleLines       0.0
InternetService     0.0
OnlineSecurity      0.0
OnlineBackup        0.0
DeviceProtection    0.0
TechSupport         0.0
StreamingTV         0.0
StreamingMovies     0.0
Contract            0.0
PaperlessBilling    0.0
PaymentMethod       0.0
MonthlyCharges      0.0
TotalCharges        0.0
Churn               0.0
dtype: float64

In [14]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder()

In [29]:
ds['Partner_LE'] = le.fit_transform(ds['Partner'])
ds['Dependents_LE'] = le.fit_transform(ds['Dependents'])
ds['PhoneService_LE'] = le.fit_transform(ds['PhoneService'])
ds['MultipleLines_LE'] = le.fit_transform(ds['MultipleLines'])
ds['InternetService_LE'] = le.fit_transform(ds['InternetService'])
ds['OnlineSecurity_LE'] = le.fit_transform(ds['OnlineSecurity'])
ds['PaymentMethod_LE'] = le.fit_transform(ds['PaymentMethod'])
ds['Churn_LE'] = le.fit_transform(ds['Churn'])
ds.head(3)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0   CUST0000    Male              0      No        Yes      23           No   
1   CUST0001  Female              0     Yes         No      43           No   
2   CUST0002    Male              1      No         No      51          Yes   

      MultipleLines InternetService       OnlineSecurity  ... TotalCharges  \
0  No phone service              No  No internet service  ...      1146.55   
1  No phone service             DSL                  Yes  ...      4330.10   
2                No             DSL                   No  ...      4963.83   

  Churn Partner_LE Dependents_LE PhoneService_LE MultipleLines_LE  \
0    No          0             1               0                1   
1   Yes          1             0               0                1   
2   Yes          0             0               1                0   

  InternetService_LE OnlineSecurity_LE  PaymentMethod_LE  Churn_LE  
0                  2                 1                 0         0  
1                  0                 2                 3         1  
2                  0                 0                 2         1  

[3 rows x 29 columns]

In [30]:
ds.drop(columns=['Partner', 'Dependents','tenure','PhoneService','MultipleLines','InternetService',	'OnlineSecurity','MultipleLines','InternetService','OnlineSecurity','Churn'	], inplace=True)

In [31]:
ds.describe()

SeniorCitizen  MonthlyCharges  TotalCharges   Partner_LE  \
count    5880.000000     5880.000000   5880.000000  5880.000000   
mean        0.500680       70.157779   2566.813165     0.509354   
std         0.500042       28.804615   1910.017743     0.499955   
min         0.000000       20.000000     20.030000     0.000000   
25%         0.000000       45.717500   1020.217500     0.000000   
50%         1.000000       70.155000   2136.445000     1.000000   
75%         1.000000       95.457500   3767.665000     1.000000   
max         1.000000      119.990000   8589.600000     1.000000   

       Dependents_LE  PhoneService_LE  MultipleLines_LE  InternetService_LE  \
count    5880.000000      5880.000000       5880.000000         5880.000000   
mean        0.489116         0.500170          1.014116            1.015816   
std         0.499924         0.500042          0.707146            0.821087   
min         0.000000         0.000000          0.000000            0.000000   
25%         0.000000         0.000000          1.000000            0.000000   
50%         0.000000         1.000000          1.000000            1.000000   
75%         1.000000         1.000000          2.000000            2.000000   
max         1.000000         1.000000          2.000000            2.000000   

       OnlineSecurity_LE  PaymentMethod_LE     Churn_LE  
count        5880.000000       5880.000000  5880.000000  
mean            0.992687          1.507653     0.492857  
std             0.809314          1.110317     0.499991  
min             0.000000          0.000000     0.000000  
25%             0.000000          1.000000     0.000000  
50%             1.000000          2.000000     0.000000  
75%             2.000000          2.000000     1.000000  
max             2.000000          3.000000     1.000000

In [33]:
ds['Churn_LE'].unique()

array([0, 1])

In [34]:
value_counts = ds['Churn_LE'].value_counts()
total_count = len(ds)
per = (value_counts / total_count) * 100
per

Churn_LE
0    50.714286
1    49.285714
Name: count, dtype: float64

It means 50.714286 are still customers and 49.285714 are not

In [35]:
ds['SeniorCitizen'].unique()

array([0, 1], dtype=int64)

In [36]:
value_count = ds['SeniorCitizen'].value_counts()
total_count2 = len(ds)
perr = (value_count / total_count2) * 100
perr

SeniorCitizen
1    50.068027
0    49.931973
Name: count, dtype: float64

 50.068027 are senior citizens and 49.931973 are not

In [39]:
churns = ds[ds['Churn_LE'] == 1] 
churn_counts = churns['SeniorCitizen'].value_counts()
total_churns = churn_counts.sum()
churn_percentages = (churn_counts / total_churns) * 100
result = pd.DataFrame({'Count': churn_counts, 'Percentage': churn_percentages})
result.index = ['Non-Senior Citizen (0)', 'Senior Citizen (1)']
result

Count  Percentage
Non-Senior Citizen (0)   1454   50.172533
Senior Citizen (1)       1444   49.827467

In [40]:
churns = ds[ds['Churn_LE'] == 1]
not_churns = ds[ds['Churn_LE'] == 0]

churn_counts = churns['SeniorCitizen'].value_counts().reindex([0, 1], fill_value=0)
not_churn_counts = not_churns['SeniorCitizen'].value_counts().reindex([0, 1], fill_value=0)

total_churns = churn_counts.sum()
total_not_churns = not_churn_counts.sum()

churn_percentages = (churn_counts / total_churns) * 100
not_churn_percentages = (not_churn_counts / total_not_churns) * 100

result = pd.DataFrame({
    'Churn Count': churn_counts,
    'Churn %': churn_percentages,
    'Not Churn Count': not_churn_counts,
    'Not Churn %': not_churn_percentages
})

# Set index labels
result.index = ['Not Senior Citizen (0)', 'Senior Citizen (1)']
result


Churn Count    Churn %  Not Churn Count  Not Churn %
Not Senior Citizen (0)         1454  50.172533             1482    49.698189
Senior Citizen (1)             1444  49.827467             1500    50.301811

In [41]:
ds['Partner_LE'].unique()

array([0, 1])